msbuild/Xamarin.iOS.Tasks.Windows/Xamarin.iOS.Windows.props

```diff
@@ -13,8 +13,9 @@ Copyright (C) 2011-2013 Xamarin. All rights reserved.
 -->
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 	<PropertyGroup>
-		<!-- Naming convention that relies on the fact that the Microsoft.iOS.Sdk pack exists with the same version that the Windows SDK -->
-		<CoreiOSSdkDirectory>$(MSBuildThisFileDirectory.Replace('Microsoft.iOS.Windows.Sdk', 'Microsoft.iOS.Sdk'))</CoreiOSSdkDirectory>
+		<CoreiOSSdkDirectory>$(MSBuildThisFileDirectory)</CoreiOSSdkDirectory>
+		<!-- Naming convention that relies on the fact that the Microsoft.iOS.Sdk pack exists with the same version that the Windows SDK (.net6 only) -->
+		<CoreiOSSdkDirectory Condition="$(MSBuildThisFileDirectory.Contains('Microsoft.iOS.Windows.Sdk'))">$(MSBuildThisFileDirectory.Replace('Microsoft.iOS.Windows.Sdk', 'Microsoft.iOS.Sdk'))</CoreiOSSdkDirectory>
```

Comment:
not sure why you need to check with `Contains` before doing a `Replace` ?
If the original string does not contain the text then replace simply return the original string, so you're effectively doubling the `Contains` checks (as `Replace` does one)

Reply:
@spouliot this is a trick to avoid using `.Replace` on old project systems (non .net SDK). If you are in a non .net SDK installation, the iOS MSBuild content in Windows will be under `Program Files (x86)`. Because the `.Replace` function is getting and encoded value, when you do it being in a sub folder of `Program Files (x86)`, it returns `Program Files %28x86%29` instead, which makes the `UsingTask` statement to fail so the build also fails.

This doesn't happen for .net SDK because the packs are under `Program Files`. This is an issue that we need to discuss with the MSBuild team, but for now we are avoiding the replace function call in old project systems doing the above.
See the current error:

```The "Xamarin.Messaging.Build.Client.Tasks.GenerateBuildSessionId" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\iOS\Xamarin.iOS.Tasks.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\iOS\Xamarin.iOS.Tasks.dll' or one of its dependencies.```